### ISSUE TITLE
npm warning with move-file

### ISSUE THREAD

**Issue by Zamiell** [raineorshine/npm-check-updates#1237]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

---

## Steps to Reproduce

Install the `npm-check-updates` package with yarn.

You should get this warning:

```
warning npm-check-updates > pacote > @npmcli/run-script > node-gyp > make-fetch-happen > cacache > @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
```

**Comment by raineorshine** [comment]
The `@npmcli/move-file` dependency was removed from `cacache` about [one month ago](https://github.com/npm/cacache/commit/4a7382f5e6c72c59587d45167346c1b6e81a3cde) and was released in `v17.0.2`. It just hasn't bubbled up the hierarchy yet:

```
npm-check-updates@16.5.0
└─┬ pacote@15.0.6
  ├─┬ @npmcli/run-script@6.0.0
  │ └─┬ node-gyp@9.3.0
  │   └─┬ make-fetch-happen@10.2.1
  │     └─┬ cacache@16.1.3
  │       └── @npmcli/move-file@2.0.1 deduped
  └─┬ cacache@17.0.0
    └── @npmcli/move-file@2.0.1
```

The latest `cacache` was added to `make-fetch-happen` in `v11.0.1`, but `node-gyp` is still using [make-fetch-happen v10.x](https://github.com/nodejs/node-gyp/blob/ee46f9d2b56eb238ce5d8199077ce5c98bdbd64c/package.json#L28). This is currently being tracked in https://github.com/nodejs/node-gyp/issues/2756. I recommend 👍'ing and subscribing to that issue, since that's the current blocker.

I would do an [override](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) but the `cacache` upgrade is a major version change, so I don't really want to risk it for a non-critical warning.

**Comment by raineorshine** [comment]
@That-Guy977 Thanks for the heads up.

Updated and published in `v16.10.17`.

**Comment by That-Guy977** [comment]
This seems to have reappeared as an issue from a different dependency, but it seems pretty weird
```sh
npm-check-updates@16.14.6
└─┬ pacote@15.2.0 (15.2.0)
  └─┬ @npmcli/run-script@6.0.2 (^6.0.0)
    └─┬ node-gyp@9.4.1 (^9.0.0)
      └─┬ make-fetch-happen@10.2.1 (^13.0.0!?)
        └─┬ cacache@16.1.3
          └── @npmcli/move-file@2.0.1
```

**Comment by raineorshine** [comment]
@That-Guy977 It looks like `node-gyp@9.4.1` reverted back to `make-fetch-happen@10.0.3`.

https://github.com/nodejs/node-gyp/blob/v9.4.1/package.json
https://github.com/nodejs/node-gyp/blob/v9.4.0/package.json

In this commit on October 27th: https://github.com/nodejs/node-gyp/commit/82c2d64ca8288853c79c6c42b9e49a64ebd21d86

I didn't see any comment or reference to an issue that provides a reason for the revert, so you may want to open an issue on https://github.com/nodejs/node-gyp/issues.

**Comment by raineorshine** [comment]
`node-gyp@10` doesn't have this problem, so the solution is for us to upgrade `pacote` which will pull in the latest `node-gyp`. This will be a breaking change since it requires bumping the node engine.

This needs to be coordinated with upgrading the rest of the dependency that bump the node engine. I have started work in the [v17](https://github.com/raineorshine/npm-check-updates/tree/v17) branch, but haven't had time to continue working on it. If someone would like to continue the process of bumping dependencies and fixing the few regressions that occur, then we could get that released.

**Comment by Zamiell** [comment]
I decided to try helping, but after forking and cloning my fork, I was greeted with this:

```
james@1qaz MINGW64 /d/Repositories/npm-check-updates (v17)
$ npm ci

> npm-check-updates@17.0.0-0 prepare
> husky install && bash test/bun-setup.sh

husky - Git hooks installed
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for bun@1.0.8: wanted {"os":"darwin,linux","arch":"arm64,x64"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin,linux
npm ERR! notsup Valid Arch:  arm64,x64
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\james\AppData\Local\npm-cache\_logs\2023-11-03T13_47_40_401Z-debug-0.log

added 790 packages, and audited 791 packages in 8s
```

AFAIK Bun isn't supported on Windows machines. I understand that Bun is fast, but by using it in this project, you are shutting out all the Windows users in the world who want to collaborate with you and help, and Windows is the most popular operating system in the world.

**Comment by raineorshine** [comment]
Hi @Zamiell. I absolutely I agree. Can you open a new issue for this? I'd be happy to respond there.

**Comment by raineorshine** [comment]
Published in `v17.0.0`.

`move-file` is no longer in the dependency tree.
